frameworklauncher/src/test/java/com/microsoft/frameworklauncher/applicationmaster/SelectionManagerTest.java

```diff
@@ -156,6 +156,43 @@ public void testSelectionManager() throws Exception {
     Assert.assertEquals(result.getNodeHosts().get(0), "node6");
     sm.addNode(node6);
 
+    //test packing logic
```

Comment:
>//test packing logic [](start = 4, length = 20)

can you use another UT function to test the packing logic specifically? also can you explain in comments about the current resource usage for the 7 nodes and what scheduling behavior you expect? (btw, why you name the last node "node10"?